model/sharing/sharing.go

```
@@ -1056,206 +1083,423 @@ func CheckSharings(inst *instance.Instance) ([]map[string]interface{}, error) {
+				"member": m.Instance,
+			})
+			return checks
 		}
 
-		if !s.Active {
-			return nil
+		var ms *Sharing
+		if ms, err = FindSharing(member, s.ID()); err != nil {
+			checks = append(checks, map[string]interface{}{
+				"id":    s.SID,
+				"type":  "missing_sharing_for_member",
+				"error": err.Error(),
+			})
+			return checks
 		}
 
-		if s.Owner {
-			for i, m := range s.Members {
-				if i == 0 || m.Status != MemberStatusReady {
-					continue
-				}
-				if s.Credentials[i-1].Client == nil {
-					checks = append(checks, map[string]interface{}{
-						"id":     s.SID,
-						"type":   "missing_oauth_client",
-						"member": i,
-						"owner":  true,
-					})
+		if ms.ReadOnly() {
+			checks = append(checks, map[string]interface{}{
+				"id":     s.SID,
+				"type":   "read_only_member",
+				"member": m.Instance,
+			})
+			return checks
+		}
+
+		memberDocs, err := FindMatchingDocs(member, *rule)
+		if err != nil {
+			checks = append(checks, map[string]interface{}{
+				"id":     s.SID,
+				"type":   "missing_matching_docs_for_member",
+				"member": m.Instance,
+				"error":  err.Error(),
+			})
+			return checks
+		}
+
+		var isMemberDiskFull bool
+		if usage, err := member.VFS().DiskUsage(); err != nil {
+			isMemberDiskFull = usage >= member.VFS().DiskQuota()
+		}
+
+		// Build a map of member docs with their owner's counterpart ids
+		memberKey := s.Credentials[i].XorKey
+		memberDocsByID := make(map[string]couchdb.JSONDoc)
+		for _, doc := range memberDocs {
+			memberDocsByID[XorID(doc.ID(), memberKey)] = doc
+		}
+
+		for _, ownerDoc := range ownerDocs {
+			if memberDoc, found := memberDocsByID[ownerDoc.ID()]; found {
+				if ownerDoc.Rev() != memberDoc.Rev() {
+					if ownerDoc.ShortRev() > memberDoc.ShortRev() && isMemberDiskFull {
+						checks = append(checks, map[string]interface{}{
+							"id":       s.SID,
+							"type":     "disk_quota_exceeded",
+							"instance": m.Instance,
+						})
+						return checks
```

When we have this error, I think we should continue to look at the other members to have a full picture.

A better way would be to have a fail fast flag, like it is implemented for `check fs`, but it is more work.

---

Right. The idea was to fail fast for the member being checked. I went a bit too fast here.
I'll take a look at the fail fast flag implementation as well.